src/App.tsx

```diff
@@ -15,47 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.NONE:
+      return visibleGoods;
+
+    case SortType.ALPHABET:
+      return [...visibleGoods].sort((good1, good2) => {
+        return good1.localeCompare(good2);
+      });
+
+    case SortType.LENGTH:
+      return [...visibleGoods].sort((good1, good2) => {
+        return good1.length - good2.length;
+      });
+
+    default:
+      break;
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getReorderedGoods(goodsFromServer, sortType);
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button', 'is-info', {
+            'is-light': sortType !== SortType.ALPHABET,
+          })}
+          onClick={() => setSortType(() => SortType.ALPHABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames('button', 'is-success', {
+            'is-light': sortType !== SortType.LENGTH,
+          })}
+          onClick={() => setSortType(() => SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(() => !isReversed)}
```

Comment:
```suggestion
          onClick={() => setIsReversed((prevState) => !prevState)}
```

Reply:
Implemented this change. Thank you for the review!